### ISSUE TITLE
TypeError: The Dirent.name getter can only be used on instances of Dirent

### ISSUE THREAD

**Issue by huseeiin** [oven-sh/bun#3370]
### What version of Bun is running?

0.6.9

### What platform is your computer?

debian 11

### What steps can reproduce the bug?

```js
const { resolve } = require("path");
const webpack = require("webpack");

const compiler = webpack({ entry: resolve(__dirname, "main.mjs") });
compiler.run((r) => {
  console.log(r);
});
```

### What is the expected behavior?

for webpack 4 to build.

### What do you see instead?

This error: TypeError: The Dirent.name getter can only be used on instances of Dirent
      at promisifiableMethods

### Additional information

I'm using the Node.js interface of webpack@4.46.0

**Comment by Jarred-Sumner** [comment]
I'm guessing it's because DirEnt is a class in Bun and Webpack probably is doing some ancient Constructor.call thing

**Comment by huseeiin** [comment]
Moved to razzle which uses webpack and its working perfectly with bun by simply running bun build/server.js

**Comment by 29decibel** [comment]
Seeing the same issue when using `@postlight/parser`:

```ts
import Parser from "@postlight/parser";
const url = process.argv[2];

Parser.parse(url).then((result) => console.log(result));
```

```bash
bun run index.ts https://github.com/oven-sh/bun
```

```
63 | function promisifiableMethods(obj, suffix, suffixRegexp, filter) {
64 |     var keys = util.inheritedDataKeys(obj);
65 |     var ret = [];
66 |     for (var i = 0; i < keys.length; ++i) {
67 |         var key = keys[i];
68 |         var value = obj[key];
                        ^
TypeError: The Dirent.name getter can only be used on instances of Dirent
      at promisifiableMethods
```

**Comment by sujayvenaik** [comment]
I am seeing a similar issue while using [bluebird](https://www.npmjs.com/package/bluebird) package. 

```
Error here 63 | function promisifiableMethods(obj, suffix, suffixRegexp, filter) {
64 |     var keys = util.inheritedDataKeys(obj);
65 |     var ret = [];
66 |     for (var i = 0; i < keys.length; ++i) {
67 |         var key = keys[i];
68 |         var value = obj[key];
```
TypeError: The Dirent.name getter can only be used on instances of Dirent
bluebird/js/main/promisify.js


-> bun -v ---- 1.0.2

**Comment by felixranesberger** [comment]
> I am seeing a similar issue while using [bluebird](https://www.npmjs.com/package/bluebird) package.
> 
> ```
> Error here 63 | function promisifiableMethods(obj, suffix, suffixRegexp, filter) {
> 64 |     var keys = util.inheritedDataKeys(obj);
> 65 |     var ret = [];
> 66 |     for (var i = 0; i < keys.length; ++i) {
> 67 |         var key = keys[i];
> 68 |         var value = obj[key];
> ```
> 
> TypeError: The Dirent.name getter can only be used on instances of Dirent bluebird/js/main/promisify.js
> 
> -> bun -v ---- 1.0.2

Since bluebird has over 12mio downloads per week, it would be awesome, if this could be fixed soon 👍

**Comment by Matoo125** [comment]
Any progress on this? I'm getting the error with @postlight/parser as well

**Comment by deemkeen** [comment]
I getting the same error in bluebird/js/main/promisify.js, but only within a ci/cd pipeline

**Comment by Strawbang** [comment]
Same error on windows with wsl try to launch [cypress](https://github.com/cypress-io/cypress) : 
<img width="775" alt="image" src="https://github.com/oven-sh/bun/assets/12730900/2f6c2db9-68bb-4f92-b09e-d0cf3a7f8977">

**Comment by Jarred-Sumner** [comment]
This was fixed in Bun v1.0.12.

To upgrade, run 
```
bun upgrade
```

**Comment by deemkeen** [comment]
@Jarred-Sumner  after the upgrade a next error occured, also in bluebird/promisify:

`TypeError: The Stats.atime getter can only be used on instances of Stats`

**Comment by Jarred-Sumner** [comment]
> @Jarred-Sumner after the upgrade a next error occured, also in bluebird/promisify:
> 
> `TypeError: The Stats.atime getter can only be used on instances of Stats`

This will be fixed in Bun v1.0.13

To upgrade (until it's released):

```
bun upgrade --canary
```

**Comment by deemkeen** [comment]
> > @Jarred-Sumner after the upgrade a next error occured, also in bluebird/promisify:
> > `TypeError: The Stats.atime getter can only be used on instances of Stats`
> 
> This will be fixed in Bun v1.0.13
> 
> To upgrade (until it's released):
> 
> ```
> bun upgrade --canary
> ```

i can confirm that everything works in canary, great stuff, team bun ftw!

**Comment by Strawbang** [comment]
> > @Jarred-Sumner after the upgrade a next error occured, also in bluebird/promisify:
> > `TypeError: The Stats.atime getter can only be used on instances of Stats`
> 
> This will be fixed in Bun v1.0.13
> 
> To upgrade (until it's released):
> 
> ```
> bun upgrade --canary
> ```

I have tried it 2 days ago but now it work great 👍
Thanks a lot !
